app/schemas.py

```
@@ -867,4 +868,86 @@ class ImageOptimizationResponse(BaseModel):
+        if 'valid_from' in values and v < values['valid_from']:
+            raise ValueError('valid_to must be after valid_from')
+        return v
+    
+    @validator('base_currency', 'target_currency')
+    def validate_currency_codes(cls, v):
+        if len(v) != 3:
+            raise ValueError('Currency code must be 3 characters')
+        return v.upper()
+    
+    @validator('base_currency', 'target_currency')
+    def validate_different_currencies(cls, v, values, **kwargs):
```

Ruff (0.14.14)

[warning] Unused method argument: `kwargs`

(ARG002)